File: Ex78ViewBinding/app/src/main/java/com/gy25m/ex78viewbinding/MyAdapter.java
package com.gy25m.ex78viewbinding;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.gy25m.ex78viewbinding.databinding.RecyclerItemBinding;

import java.util.ArrayList;

public class MyAdapter extends RecyclerView.Adapter<MyAdapter.VH> {
    Context context;
    ArrayList<ItemVO> items;

    public MyAdapter(Context context, ArrayList<ItemVO> items) {
        this.context = context;
        this.items = items;
    }

    @NonNull
    @Override
    public VH onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View itemview= LayoutInflater.from(context).inflate(R.layout.recycler_item,parent,false);
        return new VH(itemview);
    }

    @Override
    public void onBindViewHolder(@NonNull VH holder, int position) {

        holder.binding.tv.setText(items.get(position).title);
        holder.binding.iv.setImageResource(items.get(position).imgResId);
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    class VH extends RecyclerView.ViewHolder{

        //recycler_item.xml과 연결되는 바인딩 클래스 참조변수

        RecyclerItemBinding binding;
        public VH(@NonNull View itemView) {
            super(itemView);
            //이미 만든 뷰객체 안에있는 자식뷰들과 연결하기는 바인딩객체
            binding=RecyclerItemBinding.bind(itemView);

        }
    }
}

File: Ex93KotlinOpenApiNaverSearch/app/src/main/java/com/gy25m/ex93kotlinopenapinaversearch/MyAdapter.kt
package com.gy25m.ex93kotlinopenapinaversearch

import android.app.Notification.Action
import android.content.Context
import android.content.Intent
import android.net.Uri
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.core.text.HtmlCompat
import androidx.recyclerview.widget.RecyclerView.Adapter
import androidx.recyclerview.widget.RecyclerView.ViewHolder
import com.bumptech.glide.Glide
import com.gy25m.ex93kotlinopenapinaversearch.databinding.RecyclerItemBinding


class MyAdapter(var context:Context,var items:MutableList<ShoppingItem>) : Adapter<MyAdapter.VH>() {

    inner class VH(var binding: RecyclerItemBinding) : ViewHolder(binding.root) // 끝

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): VH {
        return VH(RecyclerItemBinding.inflate(LayoutInflater.from(context),parent,false))
    }

    override fun getItemCount():Int= items.size  //리턴문 한줄일때만 가능

    override fun onBindViewHolder(holder: VH, position: Int) {
        var item:ShoppingItem= items[position]

        // 제목글씨중에 HTML 태그문이 포함되어있어서 지저분함.. 이를 제거
        var title:String=HtmlCompat.fromHtml(item.title,HtmlCompat.FROM_HTML_MODE_COMPACT).toString()
        holder.binding.tvTitle.text=title
        holder.binding.tvLprice.text="${item.lprice}원"
        holder.binding.tvMall.text=item.mallName
        Glide.with(context).load(item.image).into(holder.binding.iv)

        holder.binding.root.setOnClickListener{
            // 디바이스의 인터넷앱 실행
            var intent=Intent(Intent.ACTION_VIEW)
            intent.data=Uri.parse(item.link)
            context.startActivity(intent)
        }
    }
}